Edited topics (#91)
HCL-TECH-SOFTWARE/digital-experience: docs/get_started/plan_deployment/traditional_deployment/cluster_consideration/clus_maint.md

# Cluster maintenance

> Maintaining HCL Portal in a cluster typically means applying fix packs and interim fixes or updating the software release level on each node in the cluster.

Reviewer: Maintaining HCL DX in a cluster typically means applying fix packs and interim fixes or updating the software release level on each node in the cluster.

Reviewer 2: Maybe let's leave the changing of HCL Portal to HCL DX to [DXQ-33845](https://jira.cwp.pnp-hcl.com/browse/DXQ-33845)?

> Fixes are considered "minor" if they don’t update the underlying HCL Portal databases or require version upgrades to supporting software such as databases servers or IBM® WebSphere® Application Server.

Reviewer: Fixes are considered "minor" if they do not update the underlying HCL DX databases or require version upgrades to support software such as databases servers or IBM® WebSphere® Application Server.

Most HCL Digital Experience service packs are not considered minor and might require the use of a separate installation procedure to ensure 24x7 availability.

> Instructions for applying maintenance service to an HCL Portal cluster are provided with the service package. Before you apply maintenance, analyze user impact. Ensure that you can provide uninterrupted service \(also referred to as 24x7 availability\), even during the maintenance phase.

Reviewer: Instructions to apply maintenance service to an HCL DX cluster are provided with the service package. Before you apply maintenance, analyze user impact. Ensure that you can provide uninterrupted service (also referred to as 24x7 availability), even during the maintenance phase.

> Instructions for applying maintenance service to an HCL Portal cluster are provided with the service package. Before you apply maintenance, analyze user impact. Ensure that you can provide uninterrupted service \(also referred to as 24x7 availability\), even during the maintenance phase.

Reviewer 2: ```suggestion
Instructions for applying maintenance service to an HCL Portal cluster are provided with the service package. Before you apply maintenance, analyze user impact. Ensure that you can provide uninterrupted service (also referred to as 24x7 availability), even during the maintenance phase.
```
We can leave "Instructions for applying maintenance service.." as is.

!!! note

>     If you don't use horizontal scaling in your environment, any fix that requires restarting equipment results in a temporary outage for your users. Existing 24x7 installation procedures do not apply to these environments.

Reviewer: If horizontal scaling in your environment is not used, any fix that requires to restart equipment results in a temporary outage for your users. Existing 24x7 installation procedures do not apply to these environments.

Reviewer 2: Retain the active voice

>     If you don't use horizontal scaling in your environment, any fix that requires restarting equipment results in a temporary outage for your users. Existing 24x7 installation procedures do not apply to these environments.

Reviewer 2: ```suggestion
    If you do not use horizontal scaling in your environment, any fix that requires restarting equipment results in a temporary outage for your users. Existing 24x7 installation procedures do not apply to these environments.
```
Use active voice, but avoid using contractions

!!! important

>     You do not have to remove the node from the cluster when you apply fixes. If you do remove a node, you might be unable to add the node back to the cluster.

Reviewer 2: ```suggestion
    You do not have to remove the node from the cluster when you apply fixes. If you remove a node, you might be unable to add the node back to the cluster.
```

## Minor fixes

Apply all minor fixes on each cluster node. Use the installation instructions that come with the fix. When you apply minor fixes that might update previously deployed enterprise applications<!-- How would users determine that a fix might update in this manner? -->, first turn off the auto-synchronization feature of the deployment manager. After you deploy the fix on all cluster nodes, you can force a manual synchronization with the deployment manager to ensure that all updates are synchronized on the nodes. Then, you can enable the auto-synchronization feature again.

> If the documentation associated with the minor fix requires that you restart HCL Portal or WebSphere Application Server, apply the minor fix one node at a time. This process enables other nodes to continue to provide service to users. However, if the fix requires an update to the HCL Portal databases, you might be required to stop the cluster before you apply the fix. If so, use a procedure that ensures 24x7 availability.

Reviewer: If the documentation associated with the minor fix requires that you restart HCL DX or WebSphere Application Server, apply the minor fix one node at a time. This process enables other nodes to continue to provide service to the users. However, if the fix requires an update to the HCL DX databases, you might be required to stop the cluster before you apply the fix. In that case, use a procedure that ensures 24x7 availability.

## Fixes and service packs

> You can approach installing service packs into an HCL Portal clustered environment several ways. Installing a service pack involves modifying the IP sprayer to remove a cluster from receiving user requests. By using this process, you offer the system time to finish handling existing user sessions and to upgrade that cluster on all nodes. After verification testing assures that the upgraded cluster is operates, it can start receiving production traffic while the next cluster is taken offline and goes through the upgrade process. This approach preserves complete 24x7 availability during the upgrade process.

Reviewer: You can approach installing service packs into an HCL DX clustered environment several ways. Installing a service pack involves to modify the IP sprayer to remove a cluster from receiving user requests. By using this process, you offer the system time to finish handling existing user sessions and to upgrade that cluster on all nodes. After verification testing assures that the upgraded cluster is operational, it can start receiving production traffic while the next cluster is taken offline and goes through the upgrade process. This approach preserves complete 24x7 availability during the upgrade process.

> You can approach installing service packs into an HCL Portal clustered environment several ways. Installing a service pack involves modifying the IP sprayer to remove a cluster from receiving user requests. By using this process, you offer the system time to finish handling existing user sessions and to upgrade that cluster on all nodes. After verification testing assures that the upgraded cluster is operates, it can start receiving production traffic while the next cluster is taken offline and goes through the upgrade process. This approach preserves complete 24x7 availability during the upgrade process.

Reviewer 2: ```suggestion
You can approach installing service packs into an HCL Portal clustered environment in several ways. Installing a service pack involves modifying the IP sprayer to remove a cluster from receiving user requests. By using this process, you offer the system time to finish handling existing user sessions and to upgrade that cluster on all nodes. After verification testing assures that the upgraded cluster is operational, it can start receiving production traffic while the next cluster is taken offline and goes through the upgrade process. This approach preserves complete 24x7 availability during the upgrade process.
```

> A separate document is available that describes the process of installing HCL Digital Experience service packs \(fix packs\) into an existing cluster while you maintain 24x7 availability. To briefly summarize this procedure, you remove a node or set of nodes from the flow of user traffic by configuring the IP sprayer and web server. Then, you upgrade the node with the service packs. After the upgrade is complete, return the node or set of nodes to the flow of user traffic. Repeat the procedure with the next node or set of nodes. Repeat this process until all nodes in the cluster are upgraded.

Reviewer 2: ```suggestion
A separate document is available that describes the process of installing HCL Digital Experience service packs \(fix packs\) into an existing cluster while you maintain 24x7 availability. To briefly summarize this procedure, you remove a node or set of nodes from the flow of user traffic by configuring the IP sprayer and web server. Then, upgrade the node with the service packs. After the upgrade is complete, return the node or set of nodes to the flow of user traffic. Repeat the procedure with the next node or set of nodes. Repeat this process until all nodes in the cluster are upgraded.
```
